docs/rspec.md

```diff
@@ -0,0 +1,96 @@
+When you provide a client to remote API, you would provide some means for its users to test their operations.
+
+Surely, they could use [webmock] to check the ultimate requests that are sent to the server. But doing this, they would inadvertedly specify not their own code, but your client's code too. What do they actually need is a means to check calls of your client's operations. This way they would back on correctness of your client, and take its interface as an endpoint.
+
+For this reason, we support a special RSpec matcher `perform_operation`. It checks, what operations are called via evil-client, and what options are used in there.
```

Comment:
What about adding some support for integration tests too? I mean something like that:

```ruby
before { stub_operation(CatClient, "cats.fetch") }

# or

before { stub_operation(CatClient, /cats\..*/) }

# or with return value
before { stub_operation(CatClient, /.*/).and_return(double("Cat")) }

# or with constraints
before { stub_operation(CatClient, "cats.fetch").with(token: "valid") }

# etc
```

Now I'm testing evil-client integration the following way:

```ruby
let(:candidates) do 
  double('candidates').tap do |stub|
    allow(client).to receive(:candidates).and_return(stub)
  end
end

it "works" do
  expect(candidates).to receive(:create).with(
        ...
   ).and_return(double('response', id: 'checkr-123'))

  # check some code that handles the response
end
```